### PR TITLE
Neo4j python samples and README

### PR DESCRIPTION
**Pull request by drumcircle**
Initial PR feedback requested.

### CODE REVIEW

File: notebooks/python/neo4j/README.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+![API selector truth table](images/neo4j_dataproc_api_selector.png)
+
+The GDS client API allows executing GDS function calls natively as well as any Cypher statement using ``run_cypher``.  Calling GDS natively function uses slightly different object syntax than writing out Cypher strings using the gds client.
+
+When working distributed data on Spark cluster, the Neo4j Spark Connector provides read and write support like other Spark Connectors.  Although the Spark Connector supports GDS calls using the Cypher ``CALL`` syntax, the best practice is to instantiate the GDS or python client as well to handle script calls more elegantly.
+
+## Using the GDS client
+The GDS api is Neo4j's newest, most elegant and compact python API.  Usage requires importing a single library and instantiating an object on one line.
+
+    from graphdatascience import GraphDataScience
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:52]: Please surround with code block

File: notebooks/python/neo4j/README.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+    neo4j_url = "bolt://<server>:7687"
+    neo4j_user = "<user>"
+    neo4j_password = "<password>"
+
+    gds = GraphDataScience(neo4j_url, auth=(neo4j_user, neo4j_password))
+
+Thereafter, executing Cypher statements of any kind simply requires calling:
+
+    gds.run_cypher('<Cypher statement>')
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:62]: Please surround with code block

File: notebooks/python/neo4j/README.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+For GDS functions, invocation is fluent such as:
+
+    g_msa, result = gds.graph.project(
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:66]: Please surround with code block, trailing back-slashes are needed for multi-line statement

File: notebooks/python/neo4j/README.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+            "logPopulation",
+            "logMedianHouseholdIncome",
+            "logMedianHomePrice",
+            "logPercentOver25WithBachelors"])
+
+In this case, `g_msa` is a handle to the in-memory projection.
+
+Writing data back from the in-memory graph is simple:
+
+    gds.graph.writeRelationship(
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:80]: Please surround with code block, trailing back-slashes are needed for multi-line statement

File: notebooks/python/neo4j/README.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    )
+
+Return data structures from the GDS client are well documented.
+
+## Spark Connector client
+On DataProc in the current release, global variables applied in `SparkSession.builder` such as `neo4j.url` do not stick between method invocations.
+
+To read records (in parallel) from Neo4j into a Spark DataFrame, execute a command like this:
+
+    (neo4jDf=spark.read.format("org.neo4j.spark.DataSource")
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:93]: Please surround with code block, trailing back-slashes are needed for multi-line statement.

Also - why is the statement surrounded with parenthesis?

File: notebooks/python/neo4j/README.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+        .option("labels", "MSA")
+        .option("url", neo4j_url)  
+        .option("authentication.type", "basic")
+        .option("authentication.basic.username", neo4j_user)
+        .option("authentication.basic.password", neo4j_password)
+        .load())
+
+Writing records (single threaded) to Neo4j from a Spark DataFrame can be executed like this:
+
+    (castBigqueryDf.write
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:103]: Please surround with code block, trailing back-slashes are needed for multi-line statement.

Also - why is the statement surrounded with parenthesis?

File: notebooks/python/neo4j/README.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+BigQuery fully supports Neo4j datatypes but casting may be necessary if the native types are mismatched.  
+
+## Big Query
+Also, BigQuery users know that plain SQL queries require `viewsEnabled` and `materializationDataset` options to be set.  
+
+Every time a query is executed, intermediary results will be cached in a temporary table in the schema (BigQuery calls it a “dataset”).
+
+That table will be reaped after 1 day by default, more frequently if overridden.
+
+    spark.conf.set("viewsEnabled","true")
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:187]: Please surround with code block

File: notebooks/python/neo4j/README.md [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+        ,percentOver25WithBachelors
+        ,medianHomePrice
+        ,medianHouseholdIncome
+        FROM `neo4jbusinessdev.census.msa_demos`
+    """
+    bigqueryDf = spark.read.format("bigquery").option("query",msa_query).load()
+
+However, materialization is not necessary when using the table (vs query) option because a BigQuery likes to keep around result caches but a table (SELECT * FROM <table>) doesn’t need a copy of itself!  For example:
+
+    msa_table="""
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:204]: Please surround with code block

File: notebooks/python/neo4j/README.md [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+        neo4jbusinessdev.census.msa_demos
+    """
+    tableBqDf = spark.read.format("bigquery").option("table",msa_table).load()
+
+Should derivative DataFrames then be required, secondary filtering can be done using Spark SQL as shown:
+
+    tableBqDf.createOrReplaceTempView("msa_df")
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:211]: Please surround with code block

File: notebooks/python/neo4j/README.md [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+    bigqueryPostQueryDf = spark.sql("SELECT * FROM msa_df WHERE population > 1000")
+
+After enriching features in a graph, it may be desirable to write them back to a relational source for use downstream in an AI/ML pipeline, or with business intelligence clients like Tableau, Looker, or PowerBI.
+
+With BigQuery, a temporary Google Cloud Storage (GCS) bucket is required.  Otherwise, everything is straight-forward in the Spark DataFrame way.
+
+    # tmp_working_bucket
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:218]: Please surround with code block and fix indentation

Also - why is the statement surrounded with parenthesis?

File: notebooks/python/neo4j/README.md [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+    floatCastSelectExpr=["NAME as name","cast(POPULATION as long) population",\
+    "cast(PERCENTOVER25WITHBACHELORS as float) percentOver25WithBachelors",\
+    "cast(MEDIANHOMEPRICE as int) medianHomePrice",\
+    "cast(MEDIANHOUSEHOLDINCOME as int) medianHouseholdIncome"]
+
+    castedDf = rawDf.selectExpr(floatCastSelectExpr)
+
+After down-casting doubles and decimals which Neo4j does not support, to floats or integers which it does, populating Neo4j nodes from a Spark DataFrame is trivial, and identical regardless of source.
+
+    (castedDf.write
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:167]: Please surround with code block, trailing back-slashes are needed for multi-line statement.

Also - why is the statement surrounded with parenthesis?

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: notebooks/python/neo4j/README.md [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+    .mode("overwrite")
+    .option("table","neo4jbusinessdev.census.msa_demo_similarity")
+    .save())
+
+## Why it matters
+Nathan says that he became a graph evangelist when he realized how much code could be avoided by persisting and querying data in Neo4j rather than re-hydrating intermediate results over and over again and using custom algorithms to analyze it.  Once the data is in Neo4j, it is also possible to leverage over 60 graph algorithms simply and declaratively.  Collaborating on a shared database is valuable.  Deleting in-memory projections is as easy as creating them.  It is easy to source data from files or relational databases, and to write back enriched features to original sources or to other targets.  Exporting graph embeddings to AI/ML systems is just another graph data science function (not shown here).
+
+Increasing demands on data scientists require higher productivity tools.  Most data scientists are still rolling their own algorithms or using “[free code](https://wiki.python.org/moin/PythonGraphLibraries)” for the job.  Others have already discovered that adding Neo4j to their python data science tool-belt offers these benefits:
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:228]: Please rephrase or remove the second sentence

File: notebooks/python/neo4j/README.md [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+
+<li>Persisting data across runs</li>
+<li>Easy querying of graph structures using Cypher query language</li>
+<li>Executing more complex algorithms faster thanks to constant-time O(1) query behavior</li>
+<li>Participating in a community of over 200,000 graph experts</li>
+<li>Sharing graph datasets between processes and team members</li>
+<li>Running more than 60+ graph data science algorithms and ML  in memory</li>
+<li>Exporting graph embeddings to AI/ML systems</li>
+<li>Starting right (and even for free) with the #1 graph database</li>
```
Comment (davidrabinowitz) [inline notebooks/python/neo4j/README.md:237]: Please remove the last bullet

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: notebooks/python/neo4j/README.md (hunk 5 above)
Comment (drumcircle) [reply notebooks/python/neo4j/README.md:93]: Parens are a trick to allow these multi-line statements to be handled without back-slashes at the end of the line.

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**